controllers/openstackdataplanenode_controller.go

```diff
@@ -218,9 +218,27 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
 
 	r.Log.Info("Node", "DeployStrategy", instance.Spec.DeployStrategy.Deploy, "Node.Namespace", instance.Namespace, "Node.Name", instance.Name)
 	if instance.Spec.DeployStrategy.Deploy {
+
+		// create deployIdentifier
+		if len(instance.Spec.DeployStrategy.DeployIdentifier) == 0 {
+			_, err = controllerutil.CreateOrUpdate(ctx, r.Client, instance, func() error {
+				newIdentifier := dataplanev1beta1.GenerateDeployIdentifier()
+				instance.Spec.DeployStrategy.DeployIdentifier = newIdentifier
```

Comment:
This needs to only generate the DeployIdentifier if the Deploy is toggled to true. Meaning, we only want to generate a unique value when the node is requested to be deployed by the user. Not on every future reconcile when Deploy is true. You can compare the old instance.Spec and the new to figure out if the field has been set to true and it was previously false.

This also does not need to be set inside controllerutil.CreateOrUpdate. It will get saved to when the instance is saved in the defer func.